### PR TITLE
chore: upgrade etcd version to 3.4.18 for ci

### PR DESCRIPTION
**Pull request by soulbird**
### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

Upgrading etcd version to 3.5.2 for ci. This way we can test the compatibility of apisix with the latest version of etcd

### CODE REVIEW

**Comment by shuaijinchao** [comment]
The reason for the upgrade can be supplemented in the `Description`.

**Comment by soulbird** [comment]
> The reason for the upgrade can be supplemented in the `Description`.

done

**Review by spacewander — COMMENTED** [review]
Should also update etcd client installation?

File: docs/en/latest/FAQ.md [hunk 1]
```diff
@@ -302,7 +302,7 @@ apisix:
 Please follow the troubleshooting steps described below:
 
 1. Make sure that there aren't any networking issues between Apache APISIX and your etcd deployment in your cluster.
-2. If your network is healthy, check whether you have enabled the [gRPC gateway](https://etcd.io/docs/v3.4.0/dev-guide/api_grpc_gateway/) for etcd. The default state depends on whether you used command line options or a configuration file to start the etcd server.
+2. If your network is healthy, check whether you have enabled the [gRPC gateway](https://etcd.io/docs/v3.5.2/dev-guide/api_grpc_gateway/) for etcd. The default state depends on whether you used command line options or a configuration file to start the etcd server.
```
Comment (spacewander) [inline docs/en/latest/FAQ.md:305]: The link is 404

**Comment by tokers** [comment]
FYI:

> This release is no longer recommended for production. Please use v3.4 minor until v3.5.3 is released. Read more in https://github.com/etcd-io/etcd/tree/main/CHANGELOG.

See https://github.com/etcd-io/etcd/releases/tag/v3.5.2 for details.

Shall we use `3.4.x` and give our users a hint about this?

**Comment by spacewander** [comment]
What about only updating the etcd in the CI? So we can get ready once the new etcd version is available for production.

**Comment by spacewander** [comment]
What about only updating the etcd in the CI? So we can get ready once the new etcd version is available for production.

**Comment by soulbird** [comment]
> What about only updating the etcd in the CI? So we can get ready once the new etcd version is available for production.

cc@tokers

**Comment by spacewander** [comment]
What about updating to the latest 3.4.x release: 3.4.18?
The current etcd used in the CI is released 2 years ago, so I think it is time for a upgrade.

**Comment by leslie-tsang** [comment]
> What about updating to the latest 3.4.x release: 3.4.18? The current etcd used in the CI is released 2 years ago, so I think it is time for a upgrade.

Agree

**Comment by tokers** [comment]
> What about updating to the latest 3.4.x release: 3.4.18? The current etcd used in the CI is released 2 years ago, so I think it is time for a upgrade.

Agree +1.

**Comment by moonming** [comment]
+1

Thanks,
Ming Wen
Twitter: _WenMing


Alex Zhang ***@***.***> 于2022年3月31日周四 10:55写道：

**Review by tokers — APPROVED** [review]

**Review by starsz — APPROVED** [review]

**Review by tzssangglass — APPROVED** [review]

**Pull request merged**